### DISCUSSION TITLE
Sample upstart/monit scripts

### DISCUSSION THREAD

**Discussion by subnetmarco** [Kong/kong#7212] [Ideas and feature requests]
We should provide some sample [upstart](http://upstart.ubuntu.com/)/[monit](https://mmonit.com/monit/) scripts that will detect if any of the Kong services is down and bring the system back up.

**Comment by CristianPupazan** [comment]
Any update on this? I tried to create one myself but upstart somehow ends up tracking a non existing process id.

**Comment by hutchic** [comment]
Share what you have @CristianPupazan maybe it can be improved upon / someone else can fix the process id tracking issue you ran into

**Comment by CristianPupazan** [comment]
I have used [this](https://github.com/Getsidecar/ansible-role-kong/blob/master/templates/etc/kong.init.d.j2) in the meantime to get things working.

I do not know upstart very well. Bellow is what I was trying at the time but it was not working correctly. 

```
description "managed by Ansible for kong cluster"
author "foo"
env LANG=en_GB.UTF-8

start on runlevel [2345]
stop on runlevel [016]

setuid kong
console log
expect fork

script
  exec kong start --conf "{{ kong_conf_dest }}"
end script
```
